hw12-solid/src/main/java/ru/otus/atm/domain/CashCell.java

```diff
@@ -0,0 +1,63 @@
+package ru.otus.atm.domain;
+
+/**
+ * CashCell represents a container that holds banknotes of a specific denomination
+ */
+public class CashCell {
+    private final int denomination;
+    private int count;
+
+    public CashCell(int denomination, int initialCount) {
+        if (denomination <= 0 || initialCount < 0) {
+            throw new IllegalArgumentException("Invalid denomination or count");
+        }
+        this.denomination = denomination;
+        this.count = initialCount;
+    }
+
+    /**
+     * Deposits additional banknotes into this cell
+     *
+     * @param additional the number of banknotes to add
+     */
+    public void deposit(int additional) {
+        if (additional <= 0) {
+            throw new IllegalArgumentException("Additional banknotes must be positive");
+        }
+        count += additional;
+    }
+
+    /**
+     * Withdraws banknotes from this cell
+     *
+     * @param number the number of banknotes to withdraw
+     * @return true if withdrawal is successful, false otherwise
+     */
+    public boolean withdraw(int number) {
+        if (number <= 0) {
+            throw new IllegalArgumentException("Number of banknotes must be positive");
```

Comment:
В исключение полезно передать `number`.
```suggestion
            throw new IllegalArgumentException("Number of banknotes must be positive. Requested " + number);
```